server/controllers/auth.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable consistent-return */
+import jwt from 'jsonwebtoken';
+
+import config from '../config';
+import User from '../models/User';
+
+function invalidResponse(response, message) {
+  return response.status(400).json({
+    success: false,
+    message,
+    user: null,
+  });
+}
+
+function successfulResponse(response, message, userData, token) {
+  return response.status(200).json({
+    success: true,
+    message,
+    user: { ...userData, token },
+  });
+}
+
+function getToken(userData) {
+  return jwt.sign(userData, config.JWT_SECRET_KEY, { expiresIn: '7d' });
+}
+
+function passwordPatternCheck(password) {
+  const pattern = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*_=+-]).{8,}$');
+  return pattern.test(password);
+}
+
+function getUserData(user) {
```

Comment:
What is this? You are returning the same thing.

Reply:
The object contains password as well, though it is hashed using bcrypt, I don't want to send it to the user, so destructuring the object, refactored the function though.